src/main/java/io/github/wimdeblauwe/hsbt/mvc/HtmxResponse.java

```diff
@@ -39,6 +48,34 @@ public HtmxResponse() {
 	 */
 	public HtmxResponse addTemplate(String template) {
         Assert.hasText(template, "template should not be blank");
+        if (!templates.stream().anyMatch(mav -> template.equals(mav.getViewName()))) {
+            templates.add(new ModelAndView(template));
+        }
+        return this;
+	}
+
+	/**
+	 * Append the rendered template or fragment as a resolved {@link View}.
+	 *
+	 * @param template must not be {@literal null} or empty.
```

Comment:
Does the "or empty" have any meaning here? Or was it just copy/pasted from the other comment? :) Same question for the other new method below.